### ISSUE TITLE
Regression: inner_combine_linter() throwing for singleton c() calls

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2017]
```r
lintr::lint(text = "c(exp())", linters = lintr::inner_combine_linter())
# <text>:1:1: warning: [inner_combine_linter] Combine inputs to vectorized functions first to take full advantage of vectorization, e.g., exp(c(x, y)) only runs the more expensive exp() once as compared to c(exp(x), exp(y)).
# c(exp())
# ^~~~~~~~
```

This should (maybe) trigger `unnecessary_concatenation_linter()`, but not `inner_combine_linter()`.
